Add cache buster config option

mrtbld opened the pull request:
This allows users to give a version tag or timestamp that is automatically included in the font files URLs as a query string parameter. This is useful to bust HTTP caches when deploying new versions of icon fonts.

This is different approach than @lzl124631x's P/R #39. It doesn't force any tagging scheme (timestamp, version number, whatever) since it is provided by the user. It works out of the box: no need to use a custom CSS template, just set `cacheBuster` to something (of course you can use it in custom templates too).

The default behavior is the same as before.

Review appreciated. :-)

mkhazov commented:
Looks good!
I think README.md should be updated as well.

lmartins commented:
Any chance of this being merged?

jefsnare commented:
@backflip Please contact me, I want to help with reviewing the PR's if you want to

backflip commented:
Hey @jefsnare, sounds awesome! I have added you as a collaborator.

jefsnare commented:
@backflip Cool, only write access should be nice, I can't merge or add changes. Can you provide this permission?

backflip commented:
Not sure, but you'll probably have to accept the invite first. I might be able to fine-tune permissions afterwards:

![image](https://cloud.githubusercontent.com/assets/654171/23557352/0ee3d5b0-0030-11e7-92a9-1db34aa50d98.png)

jefsnare commented:
@backflip Done, missed that email for some reason. Thank you!

jefsnare commented:
@backflip my username is jefsnare on NPM

jefsnare commented on templates/_icons.css:
```diff
@@ -1,11 +1,11 @@
-		url('<%= fontPath %><%= fontName %>.ttf') format('truetype'),
-		url('<%= fontPath %><%= fontName %>.svg#<%= fontName %>') format('svg');
+	src: url('<%= fontPath %><%= fontName %>.eot<%= cacheBusterQueryString %>');
+	src: url('<%= fontPath %><%= fontName %>.eot?<%= cacheBuster %>#iefix') format('eot'),
```
Why are you using assigned `var` `cacheBuster` instead of also using `cacheBusterQueryString ` ?

mrtbld replied:
This is because for EOT format, the IE fix needs to always have the `?` of the querystring whether or not a cache buster is used. When it is not, for other formats the `?` is ommitted but not for EOT.

This is the only difference between the two vaiables `cacheBusterQueryString` and `cacheBuster`: the former include the `?`, the latter doesn't. I used to variables for this because having `if`s in the template made the whole thing less readable (too much clutter).

(Sorry for the late response!)

mrtbld commented:
I just rebased this PR on master, and added a fixup to add documentation (and simplify code a little bit). Left the fixup so that you can see the changes but it should be squashed before merge.

However tests don't pass anymore on my computer; don't know why. The test this PR adds is still OK but the ones that was there before are not anymore for some reason. It seems to be a difference in generated font vs expected. I'm guessing a dependency changed the way it generates fonts files a bit.

Are you able to reproduce this issue?

Test output:

```
1) gulp-iconfont-css should generate SCSS file and fonts:

      Uncaught AssertionError: 'wOF2\u0000\u0001\u0000\u0000\u0000\u0000\u0003\u0000\u0000\u000b\u0000\u0000\u0000\u0000\u0006�\u0000\u0000\u0002�\u0000\u0001\ == 'wOF2\u0000\u0001\u0000\u0000\u0000\u0000\u0002�\u0000\u000b\u0000\u0000\u0000\u0000\u0006�\u0000\u0000\u0002�\u0000\u0001\u0000
      + expected - actual

      -wOF2
             ��BV�~
      -�h�W6$

                �r:��d_aS�|���̌�R��]~`�R��<��}����˄
      -i<P	Ș�J`U�t��A��U�{��h����y��@�TE��u�k �f�	��\�
                                                                     {�mG7�|�<�L7N�O_��j�{�M�:;O���OW����؀;���4��l�i�}��Ȋ=d��X�h�!@�$���~ll 8;�ݸ
                                                                                                                                                       ��6a9��\
ST;��8G�@�y� 0 ��v��u*5��X�����#� ���+���9��ȡa��0���jO                                                                                                 U�e��7�&4f���3
                                                              �l���������e�դN<|������a����t��GҖ�K9[6�'�TxN�%���a��s�Ŀ�����?N�ߗ���u�*W��ַ�>�PoB��^�X��U=�c�"��M����7k�1ezQ���a
    �����*����+F���v�s����Sa?���	���i��$}N�蒒5�(��<�a��A�uc�d2�C����K 9�b�%����g_&5	e�H��R��C��4����
                                                                                                                    �wI
                                                                                                                        �q)S�o[.��T!e`jF��3�6���[�b $����1rI����Rׅ�Sͥ��s)S�o[�T��T�2�4F��Y�m3z�:@��fuS
      -�9.8�$rY�.k�D�/�{�����f:�fF��-2����8u�
                                                �Q
      -
      +wOF2�
              ��BV�~
      +�h�Y6$

                �r:��.�mؓ�
      +$`9��
            Do�6ou�n�P�*�m������_]M����e�FM���H����w���SU���-/Iu��^#<������-���	`s�᧛�1��
                                                                                                 x����e�����|��!=��=�葏	���Q�;��
                                                                                                                                        ��* �ޡ]h�|��Ke���lK)7����j"iL���z�Y)�"ͨԀ�S�$�Ufn���xE����(E��S
      +�	�ᯫ�`��p�6H
                                  �^�XN܆��Ce���k=�v��Î
                                                       �n�o;���OKK��������M�{5?���p���M�Y�kq��˔�
      +G[s��GS���:�a�����ߧo�g�����P�e�����6��>��{�wg��}ߪ�����j@�|h��0kb�4�f\7%��w~-G�����;�2����u�G_)4Ë��i�9	�}�d��׊" f,pc@lhpe�\��2��A�n$�Iz�q�+%X-P�=+v����� I� ���H��d,wJ��d��%��= s�����l�&���q��y?dl�9ǌ�^qK9W(@o�v�G"
      +��
         o8JS��pL��B���?v��Ӟ�������hc)������V������5�1�%Hp�gYڈ
      +RT�2�&&8{S)�
      +{^��DO���4�5{���[En�C��ȶ��p�5�X$B
      
      at test/main.js:51:12
      at Stream.<anonymous> (node_modules/event-stream/index.js:318:20)
      at _end (node_modules/through/index.js:65:9)
      at Stream.stream.end (node_modules/through/index.js:74:5)
      at DestroyableTransform.onend (internal/streams/legacy.js:44:10)
      at node_modules/vinyl-fs/node_modules/readable-stream/lib/_stream_readable.js:965:16
      at _combinedTickCallback (internal/process/next_tick.js:73:7)
      at process._tickCallback (internal/process/next_tick.js:104:9)
```

backflip commented:
@mrtbld, I have updated the test files in master, I guess the woff packages were updated upstream.  
Thanks a bunch for this new feature!
